Backend/app/rider-platform/rider-app/Main/src/API/UI/Search.hs

```diff
@@ -155,7 +159,9 @@ oneWaySearch personId bundleVersion clientVersion req = do
   routeResponse <- Maps.getRoutes person.merchantId request
   let shortestRouteInfo = getRouteInfoWithShortestDuration routeResponse
   let longestRouteDistance = (.distance) =<< getLongestRouteDistance routeResponse
-  dSearchRes <- DOneWaySearch.oneWaySearch person merchant req bundleVersion clientVersion longestRouteDistance ((.distance) =<< shortestRouteInfo)
+  dSearchRes <- DOneWaySearch.oneWaySearch person merchant req bundleVersion clientVersion longestRouteDistance device ((.distance) =<< shortestRouteInfo)
+  DB.runTransaction do
```

Comment:
Can you remove this transaction and create search request with proper `estimatedRideDuration in DOneWaySearch.oneWaySearch` function instead?